src/components/headerMain/headerMain.tsx

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useState } from 'react';
+import './_headerMain.scss';
+import { Link, useNavigate } from 'react-router-dom';
+import IconButton from '@mui/material/IconButton';
+import ShoppingBasketIcon from '@mui/icons-material/ShoppingBasket';
+import SearchInput from '../searchInput/searchInput';
+import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
+import LocalMallIcon from '@mui/icons-material/LocalMall';
+import LoginIcon from '@mui/icons-material/Login';
+import Modal from '@mui/material/Modal';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Stack from '@mui/material/Stack';
+import AccountCircleIcon from '@mui/icons-material/AccountCircle';
+import NoAccountsIcon from '@mui/icons-material/NoAccounts';
+
+//Styles for modal from mui
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 350,
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+};
+
+const HeaderMain: React.FC = () => {
+  const navigate = useNavigate();
+  const isLogged = false;{/*Instead of this we will check from redux state, if user isLooged or no*/}
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [open, setOpen] = useState<boolean>(false);
+  const handleOpen = ():void => setOpen(true);
+  const handleClose = (): void => setOpen(false);
+  const inputChangeHandler = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    setSearchValue(e.target.value);
+  };
+
+  useEffect(() => {
+    //Here will be axios request for backend, every time when searchValue is changed
+  }, [searchValue]);
+
+  return (
+    <div className="header-container">
+      <div className="header-link">
+        <IconButton size="small">
+          <ShoppingBasketIcon fontSize="small" style={{ color: 'white' }} />
+        </IconButton>
+        <span className="header-text">Discover new courses</span>
+      </div>
+      <SearchInput
+        placeholder="Search courses"
+        value={searchValue}
+        onChange={inputChangeHandler}
+      />
+      <div className="header-list">
+        <Link to="#">{/*Will be Wishlist route*/}
+          <IconButton size="medium">
+            <FavoriteBorderIcon fontSize="medium" style={{ color: 'white' }} />
+          </IconButton>
+        </Link>
+        <Link to="#">{/*Will be Cart route*/}
+          <IconButton size="medium">
+            <LocalMallIcon fontSize="medium" style={{ color: 'white' }} />
+          </IconButton>
+        </Link>
+        {/*When user is logged, we will see his avatar. If no - link for login/registarion pages*/}
+        {isLogged ? (
+          <img />
```

Comment:
You have an 'img ' tag, but it does not have the alt attribute set. It is recommended to add alt for better accessibility and SEO.

Reply:
Thanks, I have added it)